packages/data-point/lib/reducer-object/resolve.test.js

```diff
@@ -0,0 +1,152 @@
+/* eslint-env jest */
+'use strict'
+
+const AccumulatorFactory = require('../accumulator/factory')
+const reducerFactory = require('../reducer/factory')
+const resolveObject = require('./resolve').resolve
+const createTransform = require('../transform-expression').create
+const resolveTransform = require('../transform-expression').resolve
+const FixtureStore = require('../../test/utils/fixture-store')
+
+let dataPoint
+
+beforeAll(() => {
+  dataPoint = FixtureStore.create()
+})
+
+describe('resolve#reducerObject.resolve', () => {
+  it('should return the accumulator for an empty reducer object', () => {
+    const reducer = reducerFactory.create(createTransform, {})
+
+    const accumulator = AccumulatorFactory.create({
+      value: {
+        x: {
+          y: {
+            z: 2
+          }
+        }
+      }
+    })
+
+    return resolveObject(
+      dataPoint,
+      resolveTransform,
+      accumulator,
+      reducer
+    ).then(result => {
+      expect(result.value).toEqual(accumulator.value)
+    })
+  })
+
+  it('should resolve a reducer object', () => {
+    const reducer = reducerFactory.create(createTransform, {
+      y: '$x.y',
+      zPlusOne: ['$x.y.z', acc => acc.value + 1]
+    })
+
+    const accumulator = AccumulatorFactory.create({
+      value: {
+        x: {
+          y: {
+            z: 2
+          }
+        }
+      }
+    })
+
+    return resolveObject(
+      dataPoint,
+      resolveTransform,
+      accumulator,
+      reducer
+    ).then(result => {
+      expect(result.value).toEqual({
```

Comment:
could be using snapshots instead? will not be so easy to read for a human, but will make your tests less bulky? dunno... just a thought

Reply:
In this case I think it's good to have the tests written out (instead of being snapshots), because it's easier to read the tests when you're trying to figure out the expected behavior of reducer objects.